### ISSUE TITLE
Recommendation appearing on profile before cutoff

### ISSUE THREAD

**Issue by yoavartzi** [lil-lab/recnet#249]
I see a recommendation appearing on my profile page before cutoff. I don't think this happened before, but not sure what changed recently. One thing that distinguishes this rec is that the paper was already known to the system, so it filled all the details automatically. 

<img width="287" alt="image" src="https://github.com/user-attachments/assets/0808c00b-0a93-4ef2-87f0-1551c3e4aaea">
<img width="732" alt="image" src="https://github.com/user-attachments/assets/18163856-50f3-40cf-b993-5401f9f862f9">

**Comment by swh00tw** [comment]
Yeah this might be a bug. I guess it's that the logic in profile page didn't filter out the upcoming rec, it just simply shows all your past recs in DB. I think this API need a fix. (I can filter out the upcoming rec in FE, but I think fix it in API side will make more sense.)
```ts

  @ApiOperation({
    summary: "Get recs",
    description: "Get historical recs by userId with pagination.",
  })
  @ApiOkResponse({ type: GetRecsResponse })
  @ApiBearerAuth()
  @Get()
  @UsePipes(new ZodValidationQueryPipe(getRecsParamsSchema))
  public async getRecs(@Query() dto: QueryRecsDto): Promise<GetRecsResponse> {
    const { page, pageSize, userId } = dto;
    return this.recService.getRecs(page, pageSize, userId);
  }
```
cc @joannechen1223

**Comment by yoavartzi** [comment]
So let's fix it properly on the API side. Can definitely wait for next week
when you two are back.

- Yoav

**Comment by joannechen1223** [comment]
Fixed in v1.3.3
